File: Dagger_APi/src/main/java/com/android/dagger_api/Teacher.java
package com.android.dagger_api;

import javax.inject.Inject;

public class Teacher {
    String name;
    String subject;

    @Inject
    public Teacher() {
    }

    public Teacher(String name, String subject) {
        this.name = name;
        this.subject = subject;
    }

    @Override
    public String toString() {
        return "hashCode:" + hashCode() + "  Teacher{" +
                "name='" + name + '\'' +
                ", subject='" + subject + '\'' +
                '}';
    }
}

File: Hilt/src/main/java/com/android/hilt/Teacher.kt
package com.android.hilt

import javax.inject.Inject

class Teacher @Inject constructor(var student1: Student) {
    @Inject
    lateinit var student2: Student
    var name: String? = ""
    var studentCount: Int? = 0

    override fun toString(): String {
        return "hashCode:" + hashCode() + "  Teacher(name=$name, studentCount=$studentCount,student1=$student1,student2=$student2)"
    }

}